### PR TITLE
fix(cli): Fix panic on argo template lint without argument

### LINKED ISSUE #4285: `argo template lint` command panic
## Summary 

`argo template lint` command panic

## Diagnostics

```
# argo template lint
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/argoproj/argo/cmd/argo/commands/template.ServerSideLint(0x3d36df0, 0x0, 0x0, 0x1, 0x2beb810, 0x5)
	/Users/acollins8/go/src/github.com/argoproj/argo/cmd/argo/commands/template/lint.go:39 +0x686
github.com/argoproj/argo/cmd/argo/commands/template.NewLintCommand.func1(0xc0006f7400, 0x3d36df0, 0x0, 0x0)
	/Users/acollins8/go/src/github.com/argoproj/argo/cmd/argo/commands/template/lint.go:27 +0x54
github.com/spf13/cobra.(*Command).execute(0xc0006f7400, 0x3d36df0, 0x0, 0x0, 0xc0006f7400, 0x3d36df0)
	/Users/acollins8/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:830 +0x2aa
github.com/spf13/cobra.(*Command).ExecuteC(0xc0006ec780, 0xc0000f0750, 0xc0009fbf50, 0x40576f)
	/Users/acollins8/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/acollins8/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
main.main()
	/Users/acollins8/go/src/github.com/argoproj/argo/cmd/argo/main.go:17 +0x2b
```

What version of Argo Workflows are you running? 

```
# argo version
argo: v2.11.5
  BuildDate: 2020-10-15T22:26:34Z
  GitCommit: 076bf89c4658adbd3b96050599f81424d1b08d6e
  GitTreeState: clean
  GitTag: v2.11.5
  GoVersion: go1.13.15
  Compiler: gc
  Platform: linux/amd64
```

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

### PR DESCRIPTION
**Pull request by sarabala1979**
Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [ ] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).
* [ ] I've signed the CLA.
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My builds are green. Try syncing with master if they are not. 
Fixes #4285

### CODE REVIEW

File: cmd/argo/commands/template/lint.go [hunk 1]
```diff
@@ -24,6 +24,10 @@ func NewLintCommand() *cobra.Command {
 		Use:   "lint (DIRECTORY | FILE1 FILE2 FILE3...)",
 		Short: "validate a file or directory of workflow template manifests",
 		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) == 0 {
```
Comment (alexec) [inline cmd/argo/commands/template/lint.go:27]: The absence of the check is intentional. It allows you to use this this xargs

Comment (sarabala1979) [reply cmd/argo/commands/template/lint.go:27]: I think xargs will be already accounted in args.

Comment (alexec) [reply cmd/argo/commands/template/lint.go:27]: no, I don't think so

```
find . -type f -name '*-workflow.yaml' | xargs argo lint
```

if find returns zero results - what happens?

Comment (sarabala1979) [reply cmd/argo/commands/template/lint.go:27]:
```
ls examples/aa*.yaml | xargs dist/argo template lint
ls: examples/aa*.yaml: No such file or directory
```

```
 ls examples/workflow-template/t*.yaml | xargs dist/argo template lint
WorkflowTemplate manifests validated
WorkflowTemplate manifests validated

```

Comment (alexec) [reply cmd/argo/commands/template/lint.go:27]: what about `true | xargs argo template lint`

Comment (sarabala1979) [reply cmd/argo/commands/template/lint.go:27]: true | xargs dist/argo template lint
sbalasubram@INTUL1732f3e11 ~/go/src/github.com/argoproj/argo (issue-4285) $

Comment (alexec) [reply cmd/argo/commands/template/lint.go:27]: exit code != 0?

Comment (sarabala1979) [reply cmd/argo/commands/template/lint.go:27]: true | xargs dist/argo template lint
sbalasubram@INTUL1732f3e11 ~/go/src/github.com/argoproj/argo (issue-4285) $ echo $?
0
it is returned existcode=0

Comment (alexec) [reply cmd/argo/commands/template/lint.go:27]: ok, do you want to add a test and we're good

Comment (sarabala1979) [reply cmd/argo/commands/template/lint.go:27]: I will add e2e test

**Review by alexec — APPROVED** [review]

**Pull request merged**